Core/Core/Conversations/CreateConversation.swift

```diff
@@ -72,20 +72,38 @@ public class CreateConversation: APIUseCase {
         guard let response = response else {
             return
         }
-        for conversation in response {
-            let conversation = Conversation.save(conversation, in: client)
 
+        for conversation in response {
+            let conversationResponse = Conversation.save(conversation, in: client)
             // attach the context name since it is not part of the api response
             // a ticket has been created to include the context_name in the api response
             // https://instructure.atlassian.net/browse/KNO-239
-            guard conversation.contextName == nil,
+            guard conversationResponse.contextName == nil,
                   let canvasContextID = canvasContextID,
                   let context = Context(canvasContextID: canvasContextID),
                   let course: Course = client.fetch(scope: .where("id", equals: context.id)).first
             else {
                 continue
             }
-            conversation.contextName = course.name
+            conversationResponse.contextName = course.name
+
+            InboxMessageListItem.save(
+                conversation,
+                currentUserID: AppEnvironment.shared.currentSession?.userID ?? "",
+                isSent: true,
+                contextFilter: .none,
+                scopeFilter: .sent,
+                in: client
+            )
+
+            InboxMessageListItem.save(
+                conversation,
+                currentUserID: AppEnvironment.shared.currentSession?.userID ?? "",
+                isSent: true,
+                contextFilter: Context(canvasContextID: conversation.context_code ?? ""),
+                scopeFilter: .sent,
+                in: client
+            )
```

Comment:
It's intentional because there is a separate entity for all list items. When a message has been sent, then it has to be shown in both All and the Course specific filtered list.